overlord/snapstate/snapstate.go

```diff
@@ -658,6 +658,15 @@ func validateFeatureFlags(st *state.State, info *snap.Info) error {
 }
 
 func ensureInstallPreconditions(st *state.State, info *snap.Info, flags Flags, snapst *SnapState, deviceCtx DeviceContext) (Flags, error) {
+	// if snap is allowed to be devmode via the dangerous model and it's
+	// confinement is indeed devmode, promote the flags.DevMode to true
+	if flags.ApplySnapDevMode && info.Confinement == snap.DevModeConfinement {
```

Comment:
the second part ican alos be written as info.NeedsDevMode()  which seems the style/intention around here